tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/concurrent/ParallelExecutorBase.java

```diff
@@ -97,7 +97,10 @@ public void addResult(CloseableIteration<T, QueryEvaluationException> res) {
 		}
 
 		try {
-			rightQueue.put(res);
+			if (!closed)
+				rightQueue.put(res);
+			else
+				res.close();
```

Comment:
Doesn't this introduce a possible race condition? The `closed` flag seems to signal if `rightQueue` has been closed or not, but there is no mutex guarding that access. It looks like it might be possible for `rightQueue` to be closed in between the check of the if-condition and the adding of the result. 

Also, minor editorial: can you add curly braces to your if-else statements? We try to consistently use this everywhere, even if the block is a one-liner.